packages/devtools-proxy-support/src/system-ca.ts

```diff
@@ -50,6 +51,54 @@ export function mergeCA(...args: (NodeJSCAOption | undefined)[]): string {
   return [...ca].join('\n');
 }
 
+const pemWithParsedCache = new WeakMap<
+  string[],
+  { pem: string; parsed: X509Certificate | null }[]
+>();
+// TODO(COMPASS-8253): Remove this in favor of OpenSSL's X509_V_FLAG_PARTIAL_CHAIN
+// See linked tickets for details on why we need this (tl;dr: the system certificate
+// store may contain intermediate certficiates without the corresponding trusted root,
+// and OpenSSL does not seem to accept that)
+export function removeCertificatesWithoutIssuer(ca: string[]): {
+  ca: string[];
+  messages: string[];
+} {
+  const messages: string[] = [];
+  let caWithParsedCerts =
+    pemWithParsedCache.get(ca) ??
+    ca.map((pem) => {
+      let parsed: X509Certificate | null = null;
+      try {
+        parsed = new X509Certificate(pem);
+      } catch (err: unknown) {
+        messages.push(
+          `Unable to parse certificate: ${
+            err && typeof err === 'object' && 'message' in err
+              ? String(err.message)
+              : String(err)
+          }`
+        );
+      }
+      return { pem, parsed };
+    });
+  pemWithParsedCache.set(ca, caWithParsedCerts);
```

Comment:
Likely inconsequential, but for my understanding, why are we storing the unfiltered certificates in the cache rather than the filtered ones? My expectation is that the result of the filter will always be the same, so we might as well cache that rather than do the filtering every time.

Reply:
Good idea! Done in 4b60c714f082f69355a7a5ed66f15980b17362f9